crates/cairo-lint-core/src/lints/double_comparison.rs

```diff
@@ -35,6 +37,12 @@ pub fn check_double_comparison(
                     binary_expr.stable_ptr().untyped(),
                     Severity::Warning,
                 ));
+            } else if is_impossible_double_comparison(&lhs_op, &rhs_op, &middle_op) {
+                diagnostics.push(create_diagnostic(
+                    IMPOSSIBLE_COMPARISON,
+                    binary_expr.stable_ptr().untyped(),
+                    Severity::Error,
+                ));
```

Comment:
this is redundant with `is_contradictory_double_comparison` can you just add your code to this function ? or even better move this into a new lint